src/item.cpp

```diff
@@ -1025,8 +1034,41 @@ bool item::stacks_with( const item &rhs, bool check_components ) const
     if( active != rhs.active ) {
         return false;
     }
-    if( item_tags != rhs.item_tags ) {
-        return false;
+    if( ignore_temperature && has_temperature() ) {
```

Comment:
Note, `has_temperature` is simply `is_food` or `is_corpse` - any item that *could* have a temperature. This whole section should be skipped if the items are not the same type - you can't stack water and oil, no matter what temperature they are. I think this is probably why the original `combine` function included the qualifier `typeId() == rhs.typeId()`.

If you're going to make such a big change to the logic of this function, I would highly recommend adding more unit tests covering what the logic is expected to do. At the very least please include some comments explaining why all these extra loops are necessary.

Reply:
type is checked at the begining of the function.